numba_dpex/kernel_api/atomic_ref.py

```diff
@@ -200,3 +200,55 @@ def exchange(self, val):
         old = self._ref[self._index]
         self._ref[self._index] = val
         return old
+
+    def compare_exchange_weak(self, expected, desired, expected_idx=0):
+        """Compares the value of the object referenced by the AtomicRef
+        against the value of ``expected[expected_idx]``.
+        If the values are equal, attempts to replace the value of the
+        referenced object with the value of ``desired``.
+        Otherwise assigns the original value of the
+        referenced object to ``expected[expected_idx]``.
+
+        Args:
+            expected : Vector containing the expected value of the
```

Comment:
Python has no type called `Vector`. Perhaps array is the right term. But, again the usage will be confusing as to why someone needs to create an array to essentially pass a single element.